### PR TITLE
Functionality for add and remove lists

### PR DESCRIPTION
**Pull request by The-Normal-One1**
### The following lists are made to add functionality to the To Do lists

- Managed to solve linters error
- Removed all hardcoded items from the tasks array.
- Created a new JavaScript file for the new functionality.
- Implemented a function for adding a new task (added a new element to the array).
- Implemented a function for deleting a task (removed an element from the array).
- Implemented a function for editing task descriptions
- implemented To Do Lists to be saved in local storage.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1,59 +1,129 @@
...
+      // update the edit todo
+      todos = todos.map((todo, index) => ({
+        ...todo,
+        description: index === editTodoId ? todoValue : todo.description,
+      }));
+
+      editTodoId = -1;
+    } else {
+      todos.push({
+        indexNum: `@${Math.floor(Math.random() * todos.length)}`,
```
Comment (Kamba56) [inline src/index.js]: - [x] Please this property should be `index` and it should be set to the value of the new array length (i.e. if you're adding a 5th task to the list, the index of that task should equal to 5). Kindly fix this please as demanded by the project requirements.

File: src/index.js [hunk 2]
```diff
@@ -1,59 +1,129 @@
...
+        description: todoValue,
+        completed: false,
+      });
+    }
+    todoInput.value = '';
+  }
 };
```
Comment (Kamba56) [inline src/index.js]: - [x] The project requirement demanded that the new functionality should be implemented in a new js file(module). Kindly move these functions to modules and import them where needed.

**Review by Kamba56 — CHANGES_REQUESTED** [review]
Hi @The-Normal-One1, 

Good job so far!!👏👏👏!👏👏👏!👏👏👏
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- [x] All linters passed!👏👏👏
- [x] Great use of Github workflow!👏👏👏

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by AmaduKamara — COMMENTED** [review]
# Status: INVALID SUBMISSION ❌❌❌

Hi @The-Normal-One1,

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that has Linter errors. Kindly fix the linter errors and resubmit for review. 👍 

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._
```

File: src/modules/tasks.js [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+      } else if (todos.some((todo) => todo.description.toUpperCase() === todoValue.toUpperCase())) {
+        alert('Todo already exists!');
+      } else {
+        if (editTodoId >= 0) {
+          // update the edit todo
+          todos = todos.map((todo, index) => ({
+            ...todo,
+            description: index === editTodoId ? todoValue : todo.description,
+          }));
+
```
Comment (Santiago220991) [inline src/modules/tasks.js]: - Kindly make sure that when you add a new task the local storage is updated. :pray: :pray: 
image

- Also please check that the index of the tasks starts at 1 not at 0 as the[ last code reviewer suggested](https://github.com/The-Normal-One1/To-Do-List/pull/2#discussion_r1013385462). :+1: :smiley: :smiley:

**Review by Santiago220991 — CHANGES_REQUESTED** [review]
Hi @The-Normal-One1,
I'm Santiago your reviewer for this PR. :wave: :wave:  

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.



 ## Highlights
:heavy_check_mark: Applied some changes required by the last reviewer.
:heavy_check_mark: No linter errors.
:heavy_check_mark: Good PR summary.

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by codepantha — APPROVED** [review]
# STATUS: APPROVED 🟢 ✔️ 

## Hi @The-Normal-One1 👋, 

Well done for implementing all the required changes 👏 
Your project is now complete and there is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**